guardian/grid
# add warning and restrictions overlay

## What does this change?

BBC only UI/UX enhancement which adds the following through config:

Adds a text message superimposed over the selected image preview tile's description
- "This image has warnings attached. It is important you check all details before use." for alert
- "This image has warnings attached. It has a restriction that requires attention." for warning
- "This image has a lease attached, read more before use" for lease attached

Adds a 100% opacity background color to the text message. Also adds a 30% opacity overlay to the selected image preview tile
- red overlay for alert
- brown/amber overlay for warning
- blue overlay for lease attached

 

<img width="1512" alt="Screen Shot 2022-11-04 at 7 29 14 PM" src="https://user-images.githubusercontent.com/79340179/200014236-40bcd8c8-8b3a-466c-b8cc-47d82d0ad461.png">

## How should a reviewer test this change?

Add the following configuration in application.conf
> warningText {
>     imagePreviewFlag {
>         alertCopy = "This image has warnings attached. It is important you check all details before use."
>         warningCopy = "This image has warnings attached. It has a restriction that requires attention."
>         leaseAttachedCopy = "This image has a lease attached, read more before use."
>     }
> }

Without the above config, the UI will behave just like before. No color overlay or warning messages on the selected preview images.

## How can success be measured?

## Who should look at this?
@guardian/digital-cms

## Tested? Documented?
